File: ruby/prime-factors/prime_factors.rb
require 'prime'

module PrimeFactors
  def self.for(number)
    primes = Prime.each(number)
    prime = primes.first
    factors = []

    while(number != 1) do
      if number % prime == 0
        factors << prime
        number = number / prime
      else
        prime = primes.next
      end
    end
    factors
  end
end

File: ecmascript/prime-factors/prime-factors.js
export default class primeFactors {
  for(number) {
    let divisor = 2,
        factors = [];

    while(number > 1) {
      if(number % divisor === 0) {
        number = number / divisor;
        factors.push(divisor);
      }
      else {
        divisor++;
      }
    }
    return factors;
  };
};
